# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsBaseChannel.cpp
# Commit: 74f99033251c
# Full Hash: 74f99033251cd6c36eb6b0b5a4713eb6b5c793fa
# Author: Jonathan Kingston <jkt@mozilla.com>
# Date: 2019-02-13 09:48:26
# Regressor Bug: 1520868
# File Overlap Count: 1
# Description:
#   Bug 1520868 - Replacing AsyncOpen2 with AsyncOpen always r=valentin
#   
#   Replacing js and text occurences of asyncOpen2
#   Replacing open2 with open
#   
# ==============================================================================

diff -r 3378fd5f4f22 -r 74f99033251c netwerk/base/nsBaseChannel.cpp
--- a/netwerk/base/nsBaseChannel.cpp	Tue Feb 12 17:58:19 2019 +0100
+++ b/netwerk/base/nsBaseChannel.cpp	Tue Feb 12 16:08:25 2019 +0000
@@ -179,12 +179,7 @@
 
   if (mOpenRedirectChannel) {
     nsresult rv = NS_OK;
-    if (mLoadInfo && mLoadInfo->GetEnforceSecurity()) {
-      MOZ_ASSERT(!mListenerContext, "mListenerContext should be null!");
-      rv = mRedirectChannel->AsyncOpen2(mListener);
-    } else {
-      rv = mRedirectChannel->AsyncOpen(mListener, mListenerContext);
-    }
+    rv = mRedirectChannel->AsyncOpen(mListener);
     NS_ENSURE_SUCCESS(rv, rv);
   }
 
@@ -214,7 +209,7 @@
   if (NS_FAILED(rv)) return rv;
 
   nsCOMPtr<nsIStreamListener> converter;
-  rv = scs->AsyncConvertData(fromType, toType, mListener, mListenerContext,
+  rv = scs->AsyncConvertData(fromType, toType, mListener, nullptr,
                              getter_AddRefs(converter));
   if (NS_SUCCEEDED(rv)) {
     mListener = converter;
@@ -293,8 +288,8 @@
 
   if (NS_FAILED(result) && mListener) {
     // Notify our consumer ourselves
-    mListener->OnStartRequest(this, mListenerContext);
-    mListener->OnStopRequest(this, mListenerContext, mStatus);
+    mListener->OnStartRequest(this, nullptr);
+    mListener->OnStopRequest(this, nullptr, mStatus);
     ChannelDone();
   }
 
@@ -576,20 +571,25 @@
 }
 
 NS_IMETHODIMP
-nsBaseChannel::Open(nsIInputStream **result) {
+nsBaseChannel::Open(nsIInputStream **aStream) {
+  nsCOMPtr<nsIStreamListener> listener;
+  nsresult rv =
+      nsContentSecurityManager::doContentSecurityCheck(this, listener);
+  NS_ENSURE_SUCCESS(rv, rv);
+
   NS_ENSURE_TRUE(mURI, NS_ERROR_NOT_INITIALIZED);
   NS_ENSURE_TRUE(!mPumpingData, NS_ERROR_IN_PROGRESS);
   NS_ENSURE_TRUE(!mWasOpened, NS_ERROR_IN_PROGRESS);
 
   nsCOMPtr<nsIChannel> chan;
-  nsresult rv = OpenContentStream(false, result, getter_AddRefs(chan));
-  NS_ASSERTION(!chan || !*result, "Got both a channel and a stream?");
+  rv = OpenContentStream(false, aStream, getter_AddRefs(chan));
+  NS_ASSERTION(!chan || !*aStream, "Got both a channel and a stream?");
   if (NS_SUCCEEDED(rv) && chan) {
     rv = Redirect(chan, nsIChannelEventSink::REDIRECT_INTERNAL, false);
     if (NS_FAILED(rv)) return rv;
-    rv = chan->Open(result);
+    rv = chan->Open(aStream);
   } else if (rv == NS_ERROR_NOT_IMPLEMENTED)
-    return NS_ImplementChannelOpen(this, result);
+    return NS_ImplementChannelOpen(this, aStream);
 
   if (NS_SUCCEEDED(rv)) {
     mWasOpened = true;
@@ -600,23 +600,23 @@
 }
 
 NS_IMETHODIMP
-nsBaseChannel::Open2(nsIInputStream **aStream) {
-  nsCOMPtr<nsIStreamListener> listener;
+nsBaseChannel::AsyncOpen(nsIStreamListener *aListener) {
+  nsCOMPtr<nsIStreamListener> listener = aListener;
+
   nsresult rv =
       nsContentSecurityManager::doContentSecurityCheck(this, listener);
-  NS_ENSURE_SUCCESS(rv, rv);
-  return Open(aStream);
-}
+  if (NS_FAILED(rv)) {
+    mCallbacks = nullptr;
+    return rv;
+  }
 
-NS_IMETHODIMP
-nsBaseChannel::AsyncOpen(nsIStreamListener *listener, nsISupports *ctxt) {
   MOZ_ASSERT(
       !mLoadInfo || mLoadInfo->GetSecurityMode() == 0 ||
           mLoadInfo->GetInitialSecurityCheckDone() ||
           (mLoadInfo->GetSecurityMode() ==
                nsILoadInfo::SEC_ALLOW_CROSS_ORIGIN_DATA_IS_NULL &&
            nsContentUtils::IsSystemPrincipal(mLoadInfo->LoadingPrincipal())),
-      "security flags in loadInfo but asyncOpen2() not called");
+      "security flags in loadInfo but doContentSecurityCheck() not called");
 
   NS_ENSURE_TRUE(mURI, NS_ERROR_NOT_INITIALIZED);
   NS_ENSURE_TRUE(!mPumpingData, NS_ERROR_IN_PROGRESS);
@@ -633,7 +633,7 @@
   }
 
   // Ensure that this is an allowed port before proceeding.
-  nsresult rv = NS_CheckPortSafety(mURI);
+  rv = NS_CheckPortSafety(mURI);
   if (NS_FAILED(rv)) {
     mCallbacks = nullptr;
     return rv;
@@ -645,7 +645,6 @@
   // this typically introduces a reference cycle between this and the listener,
   // we need to be sure to break the reference if this method does not succeed.
   mListener = listener;
-  mListenerContext = ctxt;
 
   // This method assigns mPump as a side-effect.  We need to clear mPump if
   // this method fails.
@@ -672,18 +671,6 @@
   return NS_OK;
 }
 
-NS_IMETHODIMP
-nsBaseChannel::AsyncOpen2(nsIStreamListener *aListener) {
-  nsCOMPtr<nsIStreamListener> listener = aListener;
-  nsresult rv =
-      nsContentSecurityManager::doContentSecurityCheck(this, listener);
-  if (NS_FAILED(rv)) {
-    mCallbacks = nullptr;
-    return rv;
-  }
-  return AsyncOpen(listener, nullptr);
-}
-
 //-----------------------------------------------------------------------------
 // nsBaseChannel::nsITransportEventSink
 
@@ -713,12 +700,12 @@
   if (!HasLoadFlag(LOAD_BACKGROUND)) {
     nsAutoString statusArg;
     if (GetStatusArg(status, statusArg)) {
-      mProgressSink->OnStatus(this, mListenerContext, status, statusArg.get());
+      mProgressSink->OnStatus(this, nullptr, status, statusArg.get());
     }
   }
 
   if (progress) {
-    mProgressSink->OnProgress(this, mListenerContext, progress, progressMax);
+    mProgressSink->OnProgress(this, nullptr, progress, progressMax);
   }
 
   return NS_OK;
@@ -780,7 +767,7 @@
   SUSPEND_PUMP_FOR_SCOPE();
 
   if (mListener)  // null in case of redirect
-    return mListener->OnStartRequest(this, mListenerContext);
+    return mListener->OnStartRequest(this, nullptr);
   return NS_OK;
 }
 
@@ -797,7 +784,7 @@
   mPumpingData = false;
 
   if (mListener)  // null in case of redirect
-    mListener->OnStopRequest(this, mListenerContext, mStatus);
+    mListener->OnStopRequest(this, nullptr, mStatus);
   ChannelDone();
 
   // No need to suspend pump in this scope since we will not be receiving
@@ -822,7 +809,7 @@
   SUSPEND_PUMP_FOR_SCOPE();
 
   nsresult rv =
-      mListener->OnDataAvailable(this, mListenerContext, stream, offset, count);
+      mListener->OnDataAvailable(this, nullptr, stream, offset, count);
   if (mSynthProgressEvents && NS_SUCCEEDED(rv)) {
     int64_t prog = offset + count;
     if (NS_IsMainThread()) {